qcodes_contrib_drivers/drivers/NationalInstruments/RFSG.py

```diff
@@ -0,0 +1,245 @@
+                           )
+
+        self.add_parameter(name="output_on",
+                           label="Output enabled",
+                           get_cmd=partial(self.get_attribute,
+                                           NIRFSG_ATTR_OUTPUT_ENABLED),
+                           set_cmd=partial(self.set_attribute,
+                                           NIRFSG_ATTR_OUTPUT_ENABLED),
+                           val_mapping=on_off_map(on_val=True, off_val=False),
+                           initial_value=False,
+                           )
+
+        self.add_parameter(name="pulse_mod_enabled",
+                           label="Pulse modulation enabled",
+                           get_cmd=partial(self.get_attribute,
+                                           NIRFSG_ATTR_PULSE_MODULATION_ENABLED
+                                           ),
+                           set_cmd=partial(self.set_attribute,
+                                           NIRFSG_ATTR_PULSE_MODULATION_ENABLED
+                                           ),
+                           val_mapping=on_off_map(on_val=True, off_val=False),
+                           initial_value=False,
+                           )
+
+        self.add_parameter(name="clock_source",
+                           label="Reference clock source",
+                           get_cmd=partial(self.get_attribute,
+                                           NIRFSG_ATTR_REF_CLOCK_SOURCE),
+                           set_cmd=self.set_clock_source,
+                           )
+
+        self.initiate()
+        self.connect_message()
+
+    def initiate(self):
+        """
+        Initiate signal generation. This causes the NI-RFSG device to leave
+        the Configuration state.
+        """
+        self.wrapper.Initiate(self._handle)
+
+    def abort(self):
+        """
+        Stop signal generation and return to the Configuration state.
+        """
+        self.wrapper.Abort(self._handle)
+
+    def ConfigureRF(self, frequency: float, power_level: float,
+                    initiate: bool):
+        """
+        NI-RFSG devices can only set both the frequency and power level
+        simultatneously. Convenience methods are defined below for setting
+        them individually.
+
+        NOTE: PXI-5670/5671 and PXIe-5672 devices must be in the Configuration
+        state before calling this function (by calling abort()), that is not
+        checked here.
+
+        Args:
+            frequency: frequency in Hz
+            power_level: power level in dBm
+            initiate: if True, call self.initiate after configuring, which
+                starts RF output
+        """
+        self.wrapper.ConfigureRF(self._handle, ViReal64(frequency),
+                                 ViReal64(power_level))
+        if initiate:
+            self.initiate()
+
+    def set_frequency(self, frequency: float, initiate: bool = False):
```

Comment:
is it desired to have access to frequency (and other settings) BOTH via a parameter and a method? if not, then the recommendation is to leave parameter as the only "interface", and make the method private (prepend with `_`)